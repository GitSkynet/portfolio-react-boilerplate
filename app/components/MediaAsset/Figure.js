import PropTypes from 'prop-types';
import styled from 'styled-components';

const Figure = styled.figure`
  align-items: center;
  display: flex;
  justify-content: center;
  order: ${(props) => props.isOdd ? 1 : 2};
  position: relative;
  width: 100%;
  background-color: ${(props) => props.isOdd ? '#0062a3' : '#a36200'};
  border-radius: 0.5rem;

  &::before {
    height: 0;
    left: 50%;
    position: absolute;
    top: 100%;
    transform: translateX(-50%);
    width: 0;
    content: "";
    border-bottom: 0;
    border-left: 0.75em solid transparent;
    border-top: 0.75em solid ${(props) => props.isOdd ? '#0062a3' : '#a36200'};
    border-right: 0.75em solid transparent;

    @media (min-width: 600px) {
      left: ${(props) => props.isOdd ? '100%' : 'inherit'};
      right: ${(props) => props.isOdd ? 'inherit' : '100%'};
      top: inherit;
      transform: translateX(0);
      border-bottom: 0.75em solid transparent;
      border-left: ${(props) => props.isOdd ? '0.75em solid #0062a3' : 0};
      border-top: 0.75em solid transparent;
      border-right: ${(props) => props.isOdd ? 0 : '0.75em solid #a36200'};
    }
  }

  @media (min-width: 600px) {
    width: auto;
  }
`;

Figure.propTypes = {
  isOdd: PropTypes.bool,
};

export default Figure;
