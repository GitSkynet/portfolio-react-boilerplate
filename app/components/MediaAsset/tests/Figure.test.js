import React from 'react';
import { shallow, mount } from 'enzyme';

import Figure from '../Figure';

describe('<Figure />', () => {
  it('should render and match the snapshot', () => {
    const wrapper = shallow(<Figure />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should adopt the "idOdd" prop', () => {
    const isOdd = false;
    const wrapper = mount(<Figure isOdd={isOdd} />);
    expect(wrapper.prop('isOdd')).toBe(isOdd);
  });
});
