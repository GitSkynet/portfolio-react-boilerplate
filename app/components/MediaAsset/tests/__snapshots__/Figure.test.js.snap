// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Figure /> should render and match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <figure
    className="Figure__Figure-gkjYXb kIBnOc"
/>,
  "nodes": Array [
    <figure
      className="Figure__Figure-gkjYXb kIBnOc"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Figure__Figure />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledComponent {
        "_reactInternalInstance": [Circular],
        "attrs": Object {},
        "context": Object {
          "__styled-components-stylesheet__": undefined,
          "__styled-components__": undefined,
        },
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "generatedClassName": "kIBnOc",
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <figure
          className="Figure__Figure-gkjYXb kIBnOc"
/>,
        "_debugID": 2,
        "_renderedOutput": <figure
          className="Figure__Figure-gkjYXb kIBnOc"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Figure__Figure />,
}
`;
